lib/proxy/log.rb

```diff
@@ -77,12 +77,15 @@ def initialize(app)
     end
 
     def call(env)
+      before = Time.now.to_f
       status = 500
       env['rack.logger'] = logger
       logger.info { "Started #{env['REQUEST_METHOD']} #{env['PATH_INFO']} #{env['QUERY_STRING']}" }
       logger.trace { 'Headers: ' + env.select {|k,v| k.start_with? 'HTTP_'}.inspect }
-      logger.trace { (body = env['rack.input'].read).empty? ? '' : 'Body: ' + body }
-      before = Time.now.to_f
+      if logger.trace? && env['rack.input']
```

Comment:
What about using the block variant instead of the ifs:

```ruby
    logger.trace do
        if env['rack.input'] && !(body = env['rack.input'].read).empty?
          env['rack.input'].rewind
          "Body: #{body}"
        else
          ''
        end
      end
```